solr/solrj/src/java/org/apache/solr/common/cloud/DocCollection.java

```diff
@@ -84,7 +88,10 @@ public DocCollection(String name, Map<String, Slice> slices, Map<String, Object>
    * @param zkVersion The version of the Collection node in Zookeeper (used for conditional updates).
    */
   public DocCollection(String name, Map<String, Slice> slices, Map<String, Object> props, DocRouter router, int zkVersion) {
-    super(props==null ? props = new HashMap<>() : props);
+    super(props);
+    if (props == null || props.containsKey("baseConfigSet")) {
```

Comment:
Why the name "baseConfigSet"?  And if it is in props, wouldn't we end up over-writing CONFIGNAME_PROP if it's there?

If props is null, how do we call props.put without getting an NPE?

Reply:
this was outdated